src/m365/spfx/commands/project/base-project-command.ts

```diff
@@ -33,115 +35,17 @@ export abstract class BaseProjectCommand extends AnonymousCommand {
-        project.tsConfigJson = JSON.parse(source);
-        project.tsConfigJson!.source = source;
-      }
-      catch { }
-    }
-
-    const tsLintJsonPath: string = path.join(projectRootPath, 'config/tslint.json');
-    if (fs.existsSync(tsLintJsonPath)) {
-      try {
-        const source = formatting.removeSingleLineComments(fs.readFileSync(tsLintJsonPath, 'utf-8'));
-        project.tsLintJson = JSON.parse(source);
-        project.tsLintJson!.source = source;
-      }
-      catch { }
-    }
-
-    const tsLintJsonRootPath: string = path.join(projectRootPath, 'tslint.json');
-    if (fs.existsSync(tsLintJsonRootPath)) {
-      try {
-        const source = formatting.removeSingleLineComments(fs.readFileSync(tsLintJsonRootPath, 'utf-8'));
-        project.tsLintJsonRoot = JSON.parse(source);
-        project.tsLintJsonRoot!.source = source;
-      }
-      catch { }
-    }
-
-    const writeManifestJsonPath: string = path.join(projectRootPath, 'config/write-manifests.json');
-    if (fs.existsSync(writeManifestJsonPath)) {
-      try {
-        const source = formatting.removeSingleLineComments(fs.readFileSync(writeManifestJsonPath, 'utf-8'));
-        project.writeManifestsJson = JSON.parse(source);
-        project.writeManifestsJson!.source = source;
-      }
-      catch { }
-    }
-
-    const yoRcJsonPath: string = path.join(projectRootPath, '.yo-rc.json');
-    if (fs.existsSync(yoRcJsonPath)) {
-      try {
-        const source = formatting.removeSingleLineComments(fs.readFileSync(yoRcJsonPath, 'utf-8'));
-        project.yoRcJson = JSON.parse(source);
-        project.yoRcJson!.source = source;
-      }
-      catch { }
-    }
+    this.readAndParseJsonFile(path.join(projectRootPath, 'config', 'config.json'), project, 'configJson');
+    this.readAndParseJsonFile(path.join(projectRootPath, 'config', 'copy-assets.json'), project, 'copyAssetsJson');
+    this.readAndParseJsonFile(path.join(projectRootPath, 'config', 'deploy-azure-storage.json'), project, 'deployAzureStorageJson');
+    this.readAndParseJsonFile(path.join(projectRootPath, 'package.json'), project, 'packageJson');
+    this.readAndParseJsonFile(path.join(projectRootPath, 'config', 'package-solution.json'), project, 'packageSolutionJson');
+    this.readAndParseJsonFile(path.join(projectRootPath, 'config', 'serve.json'), project, 'serveJson');
+    this.readAndParseJsonFile(path.join(projectRootPath, 'tsconfig.json'), project, 'tsConfigJson');
+    this.readAndParseJsonFile(path.join(projectRootPath, 'config', 'tslint.json'), project, 'tsLintJson');
+    this.readAndParseJsonFile(path.join(projectRootPath, 'tslint.json'), project, 'tsLintJsonRoot');
+    this.readAndParseJsonFile(path.join(projectRootPath, 'config', 'write-manifests.json'), project, 'writeManifestsJson');
+    this.readAndParseJsonFile(path.join(projectRootPath, '.yo-rc.json'), project, 'yoRcJson');
```

Comment:
Not really a comment but rather a nice job on this one. Looks way cleaner! 😄

Reply:
Thanks for the input, yes, it look good